addons/account/models/account_account.py

```diff
@@ -435,6 +441,29 @@ def _compute_account_type(self):
             closest_index = bisect_left(codes_list, account.code) - 1
             account.account_type = accounts_with_codes[account.company_id.id][codes_list[closest_index]] if closest_index != -1 else 'asset_current'
 
+    @api.depends('code')
+    def _compute_account_tags(self):
+        """
+            Compute the account tags based on the account code.
+            Search for the closest parent account code and sets the account tags according to the parent.
+            If there is no parent (e.g. the account code is lower than any other existing account code),
+            the account tags will be an empty list.
+        """
+        accounts_to_process = self.filtered(lambda r: r.code and not r.tag_ids)
+        all_accounts = self.search_read(
+            domain=[('company_id', 'in', accounts_to_process.company_id.ids)],
+            fields=['code', 'tag_ids', 'company_id'],
+            order='code',
+        )
+        accounts_with_codes = defaultdict(dict)
```

Comment:
we could extract part of the compute in a function to avoid duplicating the code with the _compute_account_type in my opinion but this way it keep the functionality splitted. Which solution is better ?